React/creating-components-2.md

```diff
@@ -0,0 +1,138 @@
+
+Move the code that renders `Student` from `App` to `StudentCollection`. Note: You will also need to move and update the `import` statement.
+
+Import and render `StudentCollection` now from the `App` component instead.
+
+[The code](https://github.com/AdaGold/react-hello-world/tree/part-4/src)
+
+
+## Use `state` for data
+
+Set up a constructor in the `StudentCollection` component which will use a variable to create a state-managed collection of student information. Don't forget your call to `super();` which is always required in a component constructor!
+
+```javascript
+constructor() {
+    super();
+
+    this.state = {
+      students: [
+        {
+          name: "State Ada",
+          email: "state-ada@ada.co"
+        },
+        {
+          name: "Cool",
+          email: "cool@ada.co"
+        }
+      ]
+    };
+  }
+```
+
+Next, we'll use a loop in our `render` function to iterate through each item in our state object and render a `Student` component for each piece of data. Remember that the `render` function needs to return a single element.
+
+```javascript
+render() {
+    let studentComponents = this.state.students.map(function(student) {
+      return <Student name={ student.name } email={ student.email }/>
+    });
+
+    return (
+      <div>
+        <h3>Students</h3>
+        { studentComponents }
+      </div>
+    )
+  }
+```
+
+Note that there are a few different ways to accomplish this same result. The approach we've taken here is to create a new variable _outside_ the `return` to construct the individual components. Then, within the return, we wrap the variable from above in a single `div` since we can only return one high-level element.
+
+Verify that you now see the data from the constructor in your browser. If you need to see what has gone wrong, take a look at [the code](https://github.com/AdaGold/react-hello-world/blob/part-5/src/components/student_collection.js).
+
+**Dive In:** Unpack the line with `return <Student...` with the person sitting next to you. Identify each piece of code on this line. Some questions to think about as you go through this activity:
+- What do you think the `studentComponents` variable looks like after this code executes?
+- How are `props` used within this code?
+- How does this code relate to the code within the `Student` component?
+- What would happen to the HTML rendered if you remove either `name` or `email`?
+
+Now let's take a look at an updated version of the diagram that we created in our last component creation lecture:
+
+![nested components](images/nested-components.png)
+<!-- https://drive.google.com/open?id=1xq5jaCrI7FGp6PG1gr-bYE1ZTvPb5PxZ -->
+
+## Some CSS
+We want our page to have a nice look & feel! Let's add some simple CSS.
+
+We can make use of the existing css file that comes from within our boilerplate React application. Let's add a new class in the `App.css` file for the student.
+
+```css
+.student {
+  padding: 10px;
+}
```

Comment:
Maybe make this something more visually distinctive, so that it's immediately obvious whether or not it worked. Something like:

```css
.student {
  width: 15em;
  border: 1px solid black;
  background-color: hsl(189, 100%, 89%);
  padding: 1em;
  margin: 1em;
}

.student-list {
  display: flex;
  justify-content: flex-start;
  flex-wrap: wrap;
}
```

(you would have to put the `.student-list` class on a div wrapping all the students but not the `<h3>` in the collection)